### PR TITLE
AP-831a Update Citizen pages so to prevent update after declaration.

### PR DESCRIPTION
**Pull request by reggieb**
## What

[Jira AP-831](https://dsdmoj.atlassian.net/browse/AP-831)

Updates most Citizen pages to use common parent `CitizenBaseController`.

Adds action to redirect to `error_path(:assessment_already_completed)` for pages on Citizen journey. Adds method to over-write this behaviour for final page, so page can be refreshed.

### CODE REVIEW

File: app/controllers/base_controller.rb [hunk 1]
```diff
@@ -1,6 +1,15 @@
 class BaseController < ApplicationController
   include Flowable
 
+  # This stops the browser caching these pages.
+  # This is done so that someone can't use the Back button to return to a users pages
+  # after they have logged out. There is a cost to page load speeds as a result
+  def set_cache_buster
```
Comment (Kalagan) [inline app/controllers/base_controller.rb]: This method still seems to be only called by the providers controllers but not the citizens controllers.
Is this an oversight or the intended use?

File: app/controllers/citizens/citizen_base_controller.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+module Citizens
+  class CitizenBaseController < BaseController
+    include ApplicationFromSession
+    before_action :authenticate_applicant!
+    before_action :check_not_complete
+
```
Comment (Kalagan) [inline app/controllers/citizens/citizen_base_controller.rb]: Don't you need to add `before_action :set_cache_buster` here?

File: app/controllers/citizens/accounts_controller.rb [hunk 3]
```diff
@@ -1,6 +1,5 @@
 module Citizens
-  class AccountsController < BaseController
-    include ApplicationFromSession
+  class AccountsController < CitizenBaseController
     before_action :authenticate_applicant!
```
Comment (Kalagan) [inline app/controllers/citizens/accounts_controller.rb]: we can remove all these now that `before_action :authenticate_applicant!` is in `CitizenBaseController`

Comment (Kalagan) [reply app/controllers/citizens/accounts_controller.rb]: this comment also applies to other controllers

File: app/controllers/citizens/citizen_base_controller.rb (hunk 2 above)
Comment (reggieb) [reply app/controllers/citizens/citizen_base_controller.rb]: Thank you! I had a niggling feeling I'd missed something there ... and that was it!

**Pull request merged**